### PR TITLE
Checker/patterns: recover on unresolved long identifiers

### PR DESCRIPTION
**Pull request by auduchinok**
Adds recovery for unresolved long identifiers in patterns, which fixes missing analysis in cases like this:

Screenshot 2024-02-16 at 14 49 14

### CODE REVIEW

File: tests/service/PatternMatchCompilationTests.fs [hunk 1]
```diff
@@ -1260,4 +1260,20 @@ let f : obj -> _ =
     assertHasSymbolUsages ["i"] checkResults
     dumpDiagnostics checkResults |> shouldEqual [
         "(5,6--5,18): Feature 'non-variable patterns to the right of 'as' patterns' is not available in F# 5.0. Please use language version 6.0 or greater."
-    ]
\ No newline at end of file
+    ]
+
+[<Test>]
+#if !NETCOREAPP
+[<Ignore("These tests weren't running on desktop and this test fails")>]
```
Comment (psfinaki) [inline tests/service/PatternMatchCompilationTests.fs:1267]: Random question - won't it work actually? I think most of the tests in this suite will pass on desktop now.
(Not asking to reenable all the tests here - wondering if this is copypaste or deliberate)

**Review by psfinaki — APPROVED** [review]
Good stuff, thank you./

File: tests/service/PatternMatchCompilationTests.fs (hunk 1 above)
Comment (abonie) [inline tests/service/PatternMatchCompilationTests.fs:1267]: I see this thing is here for other tests in this file, but do you happen to know what's this about? (why would they fail on desktop?)

Comment (auduchinok) [reply tests/service/PatternMatchCompilationTests.fs:1267]: It didn't work previously, and nowadays I just copy it as a template. 🙂

Comment (auduchinok) [reply tests/service/PatternMatchCompilationTests.fs:1267]: Something was wrong with the checking, I can't remember what exactly, unfortunately.

Comment (psfinaki) [reply tests/service/PatternMatchCompilationTests.fs:1267]: Hah, same question at the same time :eyes: 
Well anyway, if this new test works for desktop, feel free to enable it.

**Review by abonie — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Comment by abonie** [comment]
@auduchinok , could you please add release note in [docs/release-notes/.FSharp.Compiler.Service/8.0.300.md](https://github.com/dotnet/fsharp/tree/main/docs/release-notes/.FSharp.Compiler.Service/8.0.300.md) if this affects user experience (I believe it does?)

File: tests/service/PatternMatchCompilationTests.fs (hunk 1 above)
Comment (psfinaki) [reply tests/service/PatternMatchCompilationTests.fs]: Okay looks like it works everywhere, I tried and pushed, resolving, merging :)

**Pull request merged**